src/FakeItEasy.Analyzer.Shared/UnusedReturnValueAnalyzer.cs

```diff
@@ -20,10 +20,25 @@ namespace FakeItEasy.Analyzer
 #endif
     public class UnusedReturnValueAnalyzer : DiagnosticAnalyzer
     {
-        private static readonly ImmutableDictionary<string, DiagnosticDescriptor> DiagnosticsMap = CreateDiagnosticsMap();
+        private static readonly ImmutableHashSet<string> CallSpecMethods =
+            ImmutableHashSet.Create(
+                StringComparer.Ordinal,
+                "FakeItEasy.A.CallTo",
+                "FakeItEasy.A.CallTo`1",
+                "FakeItEasy.A.CallToSet`1",
+                "FakeItEasy.Fake`1.CallsTo`1",
+                "FakeItEasy.Fake`1.CallsToSet`1",
+                "FakeItEasy.Fake`1.AnyCall",
+                "FakeItEasy.ArgumentValidationConfigurationExtensions.WithAnyArguments`1",
+                "FakeItEasy.WhereConfigurationExtensions.Where`1",
+                "FakeItEasy.Configuration.IAnyCallConfigurationWithNoReturnTypeSpecified.WithReturnType`1",
+                "FakeItEasy.Configuration.IAnyCallConfigurationWithNoReturnTypeSpecified.WithNonVoidReturnType",
+                "FakeItEasy.Configuration.IArgumentValidationConfiguration`1.WhenArgumentsMatch",
+                "FakeItEasy.Configuration.IPropertySetterAnyValueConfiguration`1.To",
+                "FakeItEasy.Configuration.IWhereConfiguration`1.Where");

```

Comment:
Again, not related to the PR (I can move it to a different PR if you prefer), but the notion of "diagnostic map" made no sense in this case. It's a list, not a map, and we don't need a "diagnostic map", since this analyzer only produces one diagnostic. I don't remember exactly why I had introduced this map, but now I don't see the point, and it was reused for no reason in `NonVirtualSetupAnalyzer`.